src/rabbit_stomp_processor.erl

```diff
@@ -998,10 +998,14 @@ ensure_endpoint(source, EndPoint, Frame, Channel, State) ->
                   end},
                  {durable, false}]
         end,
-    rabbit_routing_util:ensure_endpoint(source, Channel, EndPoint, Params, State);
-
-ensure_endpoint(Direction, Endpoint, _Frame, Channel, State) ->
-    rabbit_routing_util:ensure_endpoint(Direction, Channel, Endpoint, State).
+    {_, _, Headers, _} = Frame,    
```

Comment:
Instead of having something like: `ensure_endpoint(Direction, Endpoint, Frame, Channel, State)` you can do `ensure_endpoint(Direction, Endpoint, {_, _, Headers, _} = Frame, Channel, State)`